### DISCUSSION TITLE
Passing hyperparameters to custom metrics in Autogluon Tabular

### DISCUSSION THREAD

**Discussion by SebastienZh** [autogluon/autogluon#3550] [Q&A] [answered]
I'm using a custom RMSE metric with a regularization term and want to tune the regularization hyperparameter gamma1. My custom metric is defined in CustomMetrics.py and imported.

When defining the custom metric, is there a way to pass gamma1 or other hyperparameters, so I can tune them automatically instead of manually adjusting?

I've tried passing hyperparameters at the top script level but this breaks some models during fit(). It seems ideal to pass hyperparameters only when defining the custom metric.

```
Example:

# in CustomMetrics.py

def my_rmse_1(y_true, y_pred):
    mse = sklearn.metrics.mean_squared_error(y_true, y_pred)
    num_below_0 = np.count_nonzero(y_pred <= 0)
    return gamma1*(num_below_0**2)+(mse**0.5)

# in main script

my_metric_1 = make_scorer(name='Custom_RMSE', score_func=CustomMetrics.my_rmse_1, optimum=0,
                                  greater_is_better=False)
```
Is this possible with Autogluon Tabular custom metrics? Or any suggestions for tuning hyperparameters of custom metrics during fit()?

**Comment by SebastienZh** [comment, marked as answer]
I have solved this problem by specifying the parameter _metric_kwargs_ as shown below:
```
my_metric_1 = make_scorer(name='Custom_RMSE', score_func=CustomMetrics.my_rmse, optimum=0,
                                  greater_is_better=False, metric_kwargs={"gamma":a})
```
I would recommend adding an example of passing hyperparameters to custom metrics to the documentation.
